test/graphql/resend_transaction_mutation_test.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ResendTransactionMutationTest < ActiveSupport::TestCase
+  QUERY = <<~EOS
+    mutation($id: String!, $transactionObject: JSON!, $signedTransaction: String!) {
+      resendTransaction(input: { id: $id, transactionObject: $transactionObject, signedTransaction: $signedTransaction }) {
+        watchedTransaction {
+          id
+          user {
+            address
+          }
+          transactionObject
+        }
+        errors {
+          field
+          message
+        }
+      }
+    }
+  EOS
+
+  test 'resend transaction mutation should work' do
+    old = create(:watching_transaction)
+    transaction_object = {
+      'nonce' => old.transaction_object.fetch('nonce'),
+      'transactionHash' => generate(:txhash)
+    }
+    attrs = {
+      'id' => old.id,
+      'transactionObject' => JSON.generate(transaction_object),
+      'signedTransaction' => generate(:signed_transaction)
+    }
+
+    tx_result = DaoServerSchema.execute(
+      QUERY,
+      context: { current_user: old.user },
+      variables: attrs
+    )
+
+    assert_nil tx_result['errors'],
+               'should work and have no errors'
+    assert_empty tx_result['data']['resendTransaction']['errors'],
+                 'should have no errors'
+
+    data = tx_result['data']['resendTransaction']['watchedTransaction']
+
+    assert_equal transaction_object, data['transactionObject'],
+                 'transactionObject should be the same'
+
+    assert_equal data['transactionObject']['nonce'], data['transactionObject']['nonce'],
+                 'nonce should be the same'
+  end
+
+  test 'resend transaction should fail safely' do
+    unauthorized_result = DaoServerSchema.execute(
+      QUERY,
+      context: {},
+      variables: {}
+    )
+
+    assert_not_empty unauthorized_result['errors'],
+                     'should fail without a user'
+
+    old = create(:watching_transaction)
+    transaction_object = {
+      'nonce' => old.transaction_object.fetch('nonce'),
+      'transactionHash' => generate(:txhash)
+    }
+    attrs = {
+      'id' => 'abc',
+      'transactionObject' => JSON.generate(transaction_object),
+      'signedTransaction' => generate(:signed_transaction)
+    }
+
+    invalid_group_result = DaoServerSchema.execute(
+      QUERY,
+      context: { current_user: old.user },
+      variables: attrs
+    )
+
+    assert_not_empty invalid_group_result['data']['resendTransaction']['errors'],
+                     'should fail with invalid id'
+
+    transaction_object = {
```

Comment:
You can use the factory `attrs_for(:watching_transaction)` instead of the fixed version